integration_test.sh

```diff
@@ -1,20 +1,33 @@
-#!/bin/bash
-
 # Build the binary
 make clean
 make
 
 # Start the server
 echo "Running an integration test on the server"
-echo "server { listen 8080; }" >> test_config
+echo "server { listen 8080;
+path /echo EchoHandler;
+path /static1 StaticFileHandler { root www; }
+path /static2 StaticFileHandler { root pics; } }" >> test_config
 ./webserver test_config &>/dev/null &
 
 # Send a request to the server using curl
 echo "Sending a curl request"
-curl -i -s localhost:8080 > test_output
+curl -i -s localhost:8080/echo > test_output_echo
+curl -s localhost:8080/static1/default.html > test_output_static1
+curl -s localhost:8080/static2/ducks.jpg > test_output_static2_jpg
+curl -s localhost:8080/static2/giphy.gif > test_output_static2_gif
+curl -s localhost:8080/static2/mario.jpeg > test_output_static2_jpeg
 
 # Verify the response from the server works as expected
-diff expected_output test_output > diff_of_output
+diff expected_echo_output test_output_echo > diff_echo_output
+EXIT_STATUS=$?
+diff www/default.html test_output_static1 > diff_static1_output
+EXIT_STATUS=$?
+diff pics/ducks.jpg test_output_static2_jpg > diff_static2_jpg_output
+EXIT_STATUS=$?
+diff pics/giphy.gif test_output_static2_gif > diff_static2_gif_output
+EXIT_STATUS=$?
+diff pics/mario.jpeg test_output_static2_jpeg > diff_static2_jpeg_output
 EXIT_STATUS=$?
```

Comment:
so I think the way this is written currently the if statement which follows will only check the success of the last command that's executed... we want to check whether any of the commands fail